main.py

```diff
@@ -0,0 +1,180 @@
+                    urlparser = URLParser()
+                    newurl = urlparser.parse(attr[1], baseURL)
+                    if newurl == '':
+                        pass
+                    else:
+                        self.anchors.append(newurl)
+
+    def handle_data(self, data):
+        self.data += data
+
+
+class Crawler():
+    def __init__(self, baseurl):
+        urlparser = URLParser()
+        baseurl2 = urlparser.parse(baseurl)
+        if baseurl2 not in crawled:
+            print "Starting Crawler on " + baseurl
+            print ''
+        self.baseURL = baseurl
+        self.externalURLs = []
+        self.toCrawl = []
+        self.crawl(self.baseURL)
+
+    def crawl(self, url):  # Main Function
+        urlparser = URLParser()
+        newurl = urlparser.parse(url)
+        if newurl not in crawled:
+            global baseURL
+            if baseURL in newurl:
+                headers = {}  # Initializes header
+                # Sets User Agent
+                headers["User-Agent"] = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
+                # Initializes Request
+                req = urllib2.Request(newurl, headers=headers)
+                res = urllib2.urlopen(req)  # Gets Response
+                info = res.info()  # Gets Request info
+                if str(res.code)[0] == "2":
+                    parser = Parser()
+                    parser.baseurl = self.baseURL
+                    parser.feed(res.read())
+                    f = open(
+                        "data/"+time.strftime("%Y-%m-%d-%H-%M-%S")+".txt", "w+")
+                    f.write("# "+newurl+"\n"+parser.data)
+                    f.close()
+                    print "Visited " + newurl + ","
+                    print "at " + info['date'] + "."
+                    print ''
+                    print 'Found links:'
+                    print ''
+                    for i in parser.anchors:
+                        print i
+                        self.toCrawl.append(i)
+                    print ''
+                    crawled.append(newurl)
+                    for i in self.toCrawl:
+                        if baseURL in i:
+                            global allows
+                            global disallows
+                            allowed = True
+                            for index in disallows:
+                                if type(index) == list:
+                                    yes = True
+                                    for j in index:
+                                        if j in index:
+                                            yes = False
+                                        else:
+                                            yes = True
+                                            break
+                                    allowed = yes
+                                    if allowed == False:
```

Comment:
according to pep8:
E712 comparison to False should be 'if cond is False:' or 'if not cond:'

same for line 141

Reply:
@sayamkanwar please check

Reply:
@mahansh564 please correct this style violation